### PR TITLE
Clickhouse profile mapping

### LINKED ISSUE #95: Add support for dbt-clickhouse adapter
Requested via LinkedIn comment. 
- See dbt docs [here](https://docs.getdbt.com/reference/warehouse-setups/clickhouse-setup)
- See clickhouse home page [here](https://clickhouse.com/)

### PR DESCRIPTION
**Pull request by roadan**
## Description

This PR adds Clickhouse profile mapping using a `generic` connection type. To prevent cosmos from attaching all generic connections, it uses a required field named `clickhouse` mapped to `extra.clickhouse`. 

To ensure the profile is claimed, users must add the following JSON to the extra field in the connection:
```JSON
{
    "clickhouse": "True"
}
```

## Related Issue(s)

closes #95 

## Checklist

[ x] I have made corresponding changes to the documentation (if required)
[ x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by jlaneve** [comment]
This is looking great! Question for you - it looks like there’s a recommendation to use a SQLite connection for Clickhouse with Airflow (https://github.com/bryzgaloff/airflow-clickhouse-plugin). Do you think this should use generic or SQLite connection type?

**Comment by Gaploid** [comment]
Wow! Clickhouse should defiantly support! Im voting up for this!

**Comment by epikhinm** [comment]
Sorry for intervention but I don't think that using [airflow-clickhouse-plugin](https://github.com/bryzgaloff/airflow-clickhouse-plugin)) for dbt-clickhouse is a good idea actually.
[dbt-clickhouse](https://github.com/clickHouse/dbt-clickhouse] is github repo from original company ClickHouse Inc and airflow-cosmos should use bindings from official company, with a great team behind their backs.

Hi @silentsokolov . Could somebody from your company make a decision about future of this feature? All of us thrilled to ha ve the dbt-clickhouse in airflow-cosmos :)

**Comment by silentsokolov** [comment]
I am completely in favor of this feature. More tools for working with ClickHouse - that's awesome.

I'm not part of ClickHouse company, so it's better to ask @genzgd or @guykoh

**Comment by genzgd** [comment]
I don't claim to fully understand this project :), but I think this PR correctly uses `dbt-clickhouse` and not the `airflow-clickhouse-plugin`?  Mapping values to the dbt-profile seems perfectly acceptable based on what I see.

**Comment by CorsettiS** [comment]
@jlaneve since clickhouse is has no official support on airflow it makes more sense from my perspective to use a generic http connector instead of SQLlite, which is somehow hacky. This implementation looks very solid for me as of now.

**Comment by CorsettiS** [comment]
The sqlite conn and HTTP generic conn both have the same args if I am not wrong, therefore it is not a big deal to use one or another. In my current company we use the http conn for clickhouse to avoid mixing it up. In the end it is just a naming convention.
I would advocate for keeping this PR relevant since it is easier, at least from a users perspective, to store the creds in airflow instead of in a profiles.yml file.

File: cosmos/profiles/clickhouse/user_pass.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            "port": self.conn.port or self.default_port,
+            "host": self.conn.host,
+            "secure": self.conn.extra_dejson.get("secure") or False,
+            "keepalives_idle": self.conn.extra_dejson.get("keepalives_idle"),
+            "sslmode": self.conn.extra_dejson.get("sslmode"),
+            "clickhouse": self.conn.extra_dejson.get("clickhouse"),
+            **self.profile_args,
+        }
+
+        return self.filter_null(profile)
```
Comment (tatiana) [inline cosmos/profiles/clickhouse/user_pass.py:55]: Please, could you add the `mock_profile` function & a test for it?
https://github.com/astronomer/astronomer-cosmos/blob/3786703609e69c1e8f4b2db1475fe8b6ea00a117/cosmos/profiles/base.py#L97C7-L97C7

This was introduced after this PR was created and is now required.

File: cosmos/profiles/clickhouse/user_pass.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+class ClickhouseUserPasswordProfileMapping(BaseProfileMapping):
+    """
+    Maps Airflow generic connections using user + password authentication to dbt Clickhouse profiles.
+    https://docs.getdbt.com/docs/core/connect-data-platform/clickhouse-setup
+    """
+
+    airflow_connection_type: str = "generic"
+    default_port = 9000
+
```
Comment (tatiana) [inline cosmos/profiles/clickhouse/user_pass.py:17]: Please, could you add the attribute:
```
is_community = True
```
https://github.com/astronomer/astronomer-cosmos/blob/3786703609e69c1e8f4b2db1475fe8b6ea00a117/cosmos/profiles/base.py#L33C31-L33C31

**Review by tatiana — COMMENTED** [review]
@roadan, sorry for the delay. Could you please rebase this branch and address the comments?

@jlaneve, at least three community members would like this feature; what do you think about approving it once the feedback has been addressed and tests passed? It seems valuable, it is well documented and tested.

**Comment by alexisvannier** [comment]
Would be very nice to see this PR merged. 👍

**Comment by vargacypher** [comment]
@roadan could i colaborate on Unresolved conversations ? 
We really want this feature.

**Comment by pankajastro** [comment]
Hey @roadan, It looks like we're very close to merging! Could you please rebase and add the mock_profile property? Let me know if there's anything I can do to help. thanks!

**Comment by pankajastro** [comment]
Hi @roadan, I was wondering if you had a chance to read my previous comment. It seems like this feature is very important for the community. If it's alright with you, can I take it forward from here? Thanks!

**Comment by tatiana** [comment]
@roadan @vargacypher @alexisvannier @CorsettiS @genzgd @epikhinm @Gaploid

Since this work is still relevant, and the PR got stale, @pankajastro is rebasing the original implementation in #1016, so we can get this work merged and released as part of Cosmos 1.5. If we release an alpha, would one of you be able to help us testing this feature?

**Comment by vargacypher** [comment]
I could help on tests @tatiana

**Comment by tatiana** [comment]
@vargacypher, thanks a lot; we just merged #1016 and let you know once we have an alpha with this feature!

**Pull request closed without merging**